cxx-idioms/vertical_or_horizontal_storage.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+template<size_t ...>
+struct seq { };
+
+template<size_t N, size_t ...S>
+struct gens : gens<N-1, N-1, S...> { };
+
+template<size_t ...S>
+struct gens<0, S...> {
+typedef seq<S...> type;
+};
+
+template <class ObjData>
+struct ObjAccessVertical: public ObjData
+{
+	using ObjData::ObjData;
+
+	template <size_t I>
+	auto& get_data()
+	{
+		return std::get<I>(static_cast<ObjData&>(*this));
+	}
+
+	template <size_t I>
+	auto const& get_data() const
+	{
+		return std::get<I>(static_cast<ObjData const&>(*this));
+	}
+};
+
+template <template <class Storage> class Obj, class... Args>
+struct ObjStorageVertical: public std::vector<Obj<ObjAccessVertical<std::tuple<Args...>>>>
+{ };
+
+template <class Storage>
+struct ObjAccessHorizontal
+{
+	ObjAccessHorizontal(Storage* s, size_t i):
+		storage(s),
+		idx(i)
+	{ }
+
+	ObjAccessHorizontal(ObjAccessHorizontal const& o):
```

Comment:
forward constructor to the previous constructor?
: ObjAccesHorizontal(o.storage, o.idx)